File: workspaces/default/output_demo.cell.cpp
// %% Cell 1 - HTML Output Demo
// Demonstrates the cm::views module for C++

#include <iostream>
#include <cm/views.hpp>

int main() {
    // Output formatted HTML
    cm::views::html("<h2>C++ HTML Output Demo</h2>");
    cm::views::html("<p>This demonstrates <strong>rich HTML output</strong> from C++.</p>");

    // Log messages
    cm::views::log("Processing started...");
    cm::views::log_success("Cell 1 complete!");

    return 0;
}

// %% Cell 2 - Tables
#include <iostream>
#include <vector>
#include <string>
#include <cm/views.hpp>

int main() {
    // Display a table of data
    std::vector<std::vector<std::string>> data = {
        {"Hydrogen", "H", "1.008"},
        {"Helium", "He", "4.003"},
        {"Lithium", "Li", "6.941"},
        {"Beryllium", "Be", "9.012"}
    };

    std::vector<std::string> headers = {"Element", "Symbol", "Atomic Mass"};

    cm::views::table(data, headers);
    cm::views::log("Table with 4 elements displayed");

    return 0;
}

// %% Cell 3 - Logging Levels
#include <iostream>
#include <cm/views.hpp>

int main() {
    cm::views::html("<h3>Log Level Examples</h3>");

    cm::views::log("This is a regular log message");
    cm::views::log_success("Operation completed successfully!");
    cm::views::log_warning("This is a warning message");
    cm::views::log_error("This is an error message");

    // Multiple arguments
    int value = 42;
    cm::views::log("The answer is:", value);

    return 0;
}

// %% Cell 4 - LaTeX Math with Line Height
#include <iostream>
#include <cm/symbols.hpp>

int main() {
    cm::views::html("<h3>LaTeX Math Demo</h3>");

    // Set custom line height (try: "1", "1.5", "2", "normal")
    cm::symbols::set_line_height("1.8");

    // Render equations
    cm::symbols::latex("E = mc^2");
    cm::symbols::latex("\\int_0^\\infty e^{-x} dx = 1");
    cm::symbols::equation("F = ma", "1");

    // Reset to normal
    cm::symbols::set_line_height("normal");
    cm::symbols::latex("\\nabla \\times \\mathbf{B} = \\mu_0 \\mathbf{J}");

    cm::views::log_success("LaTeX rendering complete!");

    return 0;
}

File: workspaces/default/output_demo.cell.py
# %% Cell 1 - HTML Output Demo
# Demonstrates the cm.views module for rich HTML outputs

from cm.views import html, text, log, table

# Output formatted HTML
html("<h2>Welcome to Chemical Machines!</h2>")
html("<p>This cell demonstrates <strong>rich HTML output</strong>.</p>")

# Log messages with different levels
log("Processing started...")
log("Computation complete!", level="success")

# %% Cell 2 - Tables and Data
from cm.views import html, table, log

# Display a table
data = [
    ["Hydrogen", "H", 1.008],
    ["Carbon", "C", 12.011],
    ["Nitrogen", "N", 14.007],
    ["Oxygen", "O", 15.999],
]
table(data, headers=["Element", "Symbol", "Atomic Mass"])

log("Table displayed with 4 elements")

# %% Cell 3 - Matplotlib Integration
from cm.views import savefig, log

try:
    import matplotlib.pyplot as plt
    import numpy as np

    # Create a simple plot
    x = np.linspace(0, 2 * np.pi, 100)
    plt.figure(figsize=(8, 4))
    plt.plot(x, np.sin(x), label='sin(x)')
    plt.plot(x, np.cos(x), label='cos(x)')
    plt.legend()
    plt.title('Trigonometric Functions')
    plt.xlabel('x')
    plt.ylabel('y')
    plt.grid(True, alpha=0.3)

    # Save to HTML output
    savefig()
    plt.close()

    log("Plot saved successfully!", level="success")
except ImportError:
    log("matplotlib not available - skipping plot demo", level="warning")
